platform/kobo/koreader.sh

```diff
@@ -36,14 +36,36 @@ if pkill -0 nickel ; then
 fi
 
 if [ "${FROM_NICKEL}" == "true" ] ; then
-	# Siphon a few things from nickel's env...
-	eval "$(xargs -n 1 -0 < /proc/$(pidof nickel)/environ | grep -e DBUS_SESSION_BUS_ADDRESS -e WIFI_MODULE -e PLATFORM -e WIFI_MODULE_PATH -e INTERFACE -e PRODUCT 2>/dev/null)"
-	export DBUS_SESSION_BUS_ADDRESS WIFI_MODULE PLATFORM WIFI_MODULE_PATH INTERFACE PRODUCT
+	# Detect if we were started from KFMon
+	if pkill -0 kfmon ; then
+		FROM_KFMON="true"
+	else
+		FROM_KFMON="false"
+	fi
+
+	if [ "${FROM_KFMON}" == "true" ] ; then
+		# Siphon nickel's full environment, since KFMon inherits such a minimal one...
+		for env in $(xargs -n 1 -0 < /proc/$(pidof nickel)/environ) ; do
+			export ${env}
```

Comment:
I didn't try to isolate what might be the exact reason, so it's kind of a mystery...
Granted, the env we get from udev is seriously bare, but then agin, there's nothing really scintillating in what we're siphoninh off here, so... -_-".